retriever/lib/dummy.py

```diff
@@ -0,0 +1,23 @@
+"""Re-usable dummy classes for connectio-less engine instances
```

Comment:
There's a typo `connection` not `connectio`, we can probably skip the "Re-usable", and no hyphen in "connectionless" gives:

`Dummy connection classes for connectionless engine instances`